crates/matrix-sdk/src/sliding_sync/list/mod.rs

```diff
@@ -508,11 +501,10 @@ impl SlidingSyncListInner {
     fn update_request_generator_state(&self, maximum_number_of_rooms: u32) -> Result<(), Error> {
         let mut request_generator = self.request_generator.write().unwrap();
 
-        let range_end: u32 = request_generator
-            .ranges
-            .first()
-            .map(|(_start, end)| u32::try_from(*end).unwrap())
-            .ok_or_else(|| Error::RequestGeneratorHasNotBeenInitialized(self.name.to_owned()))?;
+        let range_end: u32 =
```

Comment:
```suggestion
        let range_end: Bound =
```

Thoughts?

Reply:
Yeah I considered this too, but since here we're manipulating the concrete value and not treat it as opaque, it felt nicer to use the actual type in this function. I don't have a strong opinion here, so can use the `Bound` if you prefer.